File: app/src/main/java/com/khmelenko/lab/travisclient/adapter/viewholder/BuildViewHolder.java
package com.khmelenko.lab.travisclient.adapter.viewholder;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.khmelenko.lab.travisclient.R;
import com.khmelenko.lab.travisclient.adapter.OnListItemListener;
import com.khmelenko.lab.travisclient.widget.BuildView;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Viewholder class
 */
public class BuildViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    @Bind(R.id.item_build_card_view)
    View mParent;

    @Bind(R.id.item_build_data)
    public BuildView mBuildView;

    private final OnListItemListener mListener;

    public BuildViewHolder(View itemView, OnListItemListener mListener) {
        super(itemView);
        this.mListener = mListener;
        ButterKnife.bind(this, itemView);
        itemView.setClickable(true);
        mParent.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (mListener != null) {
            mListener.onItemSelected(getLayoutPosition());
        }
    }
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/adapter/viewholder/BuildViewHolder.kt
package com.khmelenko.lab.varis.adapter.viewholder

import androidx.recyclerview.widget.RecyclerView
import android.view.View
import com.khmelenko.lab.varis.widget.BuildView
import kotlinx.android.synthetic.main.item_build_view.view.item_build_card_view
import kotlinx.android.synthetic.main.item_build_view.view.item_build_data

/**
 * View holder for the Build data
 *
 * @author Dmytro Khmelenko
 */
class BuildViewHolder(itemView: View, private val listener: (Int) -> Unit) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

    val parent: View

    val buildView: BuildView

    init {
        itemView.isClickable = true
        parent = itemView.item_build_card_view
        parent.setOnClickListener(this)

        buildView = itemView.item_build_data
    }

    override fun onClick(view: View) {
        listener(layoutPosition)
    }
}
